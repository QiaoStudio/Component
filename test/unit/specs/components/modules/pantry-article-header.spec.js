import {
  shallowMount
} from 'dp-test'
import ComArticleHeader from 'components/modules/article-header/article-header.vue'
import sinon from 'sinon'
import _ from 'lodash'

describe('/modules/Com Article Header', () => {
  let wrapper
  let sanbox
  const title = 'Test'
  const alt = 'Test Image'
  const levels = [
    {
      text: 'Blog',
      link: 'https://www.xxxxxx.com',
      isVisible: true
    },
    {
      text: 'Budgeting',
      link: 'https://www.xxxxxx.com',
      isVisible: true
    }
  ]
  const author = {
    author: 'Walter White',
    href: 'http://xxxxxx.com',
    authorAvatar: 'https://picsum.photos/62',
    alt: 'Walter White',
    time: new Date('2018-11-22').toUTCString(),
    maxW: 40,
    maxH: 40,
    isSpanWrap: false,
    formatter: 'DD MMM YYYY'
  }

  function createWrapper(options) {
    return shallowMount(ComArticleHeader, options)
  }

  function generateWrapper(options) {
    return createWrapper(_.merge({
      propsData: {
        levels: levels,
        author: author,
        title: title,
        imageAlt: alt,
        imageUrl: 'https://picsum.photos/1218/754',
        mobileImageUrl: 'https://picsum.photos/767/474'
      }
    }, options || {}))
  }

  beforeEach(() => {
    sanbox = sinon.createSandbox()
    wrapper = generateWrapper()
  })

  afterEach(() => {
    wrapper.destroy()
    sanbox.restore()
  })

  it('when article header is initialized, article header should match snapshot', () => {
    wrapper.setProps({
      title: 'Title 1',
      imageAlt: 'New Image',
      imageUrl: 'https://picsum.photos/100/100',
      mobileImageUrl: 'https://picsum.photos/50/50'
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('when article header title is set, article header title should be shown', () => {
    let title = 'Hello World!'
    wrapper.setProps({
      title: title
    })
    expect(wrapper.find('.dp-article-header .dp-title').exists()).toBe(true)
    expect(wrapper.find('.dp-article-header .dp-title').text()).toBe(title)
  })

  it('when article header imageUrl is set, article header imageUrl should be correct', () => {
    let imageUrl = 'https://picsum.photos/40/40'
    wrapper.setProps({
      imageAlt: alt,
      imageUrl: imageUrl
    })
    expect(wrapper.find('.dp-article-header__image img').exists()).toBe(true)
    expect(wrapper.find('.dp-article-header__image img').attributes().src).toBe(imageUrl)
    expect(wrapper.find('.dp-article-header__image img').attributes().alt).toBe(alt)
  })

  it('when article header mobileImageUrl is set, article header mobileImageUrl should be correct', () => {
    let mobileImageUrl = 'https://picsum.photos/1/1'
    wrapper.vm.$screen.resize = () => true
    wrapper.vm.$screen.mobile = () => true
    wrapper = createWrapper({
      propsData: {
        levels: levels,
        author: author,
        title: title,
        imageAlt: alt,
        imageUrl: 'https://picsum.photos/1218/754',
        mobileImageUrl: mobileImageUrl
      },
      computed: {
        isDesktop: () => false
      }
    })
    expect(wrapper.find('.dp-article-header__image img').exists()).toBe(true)
    expect(wrapper.vm.$el.querySelector('.dp-article-header__image img').originalSrc).toBe(mobileImageUrl)
    expect(wrapper.find('.dp-article-header__image img').attributes().src).toBe('invis.png')
    expect(wrapper.find('.dp-article-header__image img').attributes().alt).toBe(alt)
    expect(wrapper.find('.dp-article-header__image img').classes()).toContain('lazy-loading')
  })
})
