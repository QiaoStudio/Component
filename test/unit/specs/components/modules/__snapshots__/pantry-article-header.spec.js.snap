// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/modules/Com Article Header when article header is initialized, article header should match snapshot 1`] = `
<div class="dp-article-header">
  <div>
    <dp-breadcrumbs-stub levels="[object Object],[object Object]"></dp-breadcrumbs-stub>
    <h1 class="dp-title">Title 1</h1>
    <div class="dp-theme-article">
      <dp-author-byline-stub author="Walter White" authoravatar="https://picsum.photos/62" alt="Walter White" time="Thu, 22 Nov 2018 00:00:00 GMT" maxheight="40" maxwidth="40" formatter="DD MMM YYYY" href="http://xxxxxx.com"></dp-author-byline-stub>
    </div>
  </div>
  <div>
    <div class="dp-article-header__image">
      <img src="https://picsum.photos/100/100" alt="New Image" class="lazy-loading" width="60" height="60">
      <dp-loader-stub></dp-loader-stub>
    </div>
  </div>
</div>
`;
